packages/esm-patient-attachments-app/src/camera-media-uploader/uploading-status.component.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useCallback, useEffect, Dispatch, SetStateAction, useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { showToast, FetchResponse } from '@openmrs/esm-framework';
+import { UploadedFile } from '../attachments-types';
+import styles from './uploading-status.scss';
+import { FileUploaderItem, Button, ButtonSet } from 'carbon-components-react';
+import CameraMediaUploaderContext from './camera-media-uploader-context.resources';
+
+interface UploadingStatusComponentProps {}
+
+const UploadingStatusComponent: React.FC<UploadingStatusComponentProps> = () => {
+  const { t } = useTranslation();
+  const [uploadingCompleted, setUploadingComplete] = useState(false);
+  const { filesToUpload, saveFile, setFilesToUpload, closeModal, clearData, onCompletion } =
+    useContext(CameraMediaUploaderContext);
+
+  useEffect(() => {
+    Promise.all(
+      filesToUpload.map((file, indx) =>
+        saveFile(file).then(() => {
+          showToast({
+            title: t('uploadComplete', 'Upload complete'),
+            description: `${file.fileName} ${t('uploadedSuccessfully', 'uploaded successfully')}`,
+            kind: 'success',
+          });
+          setFilesToUpload((prevfilesToUpload) =>
+            prevfilesToUpload.map((file, ind) =>
+              ind === indx
+                ? {
+                    ...file,
+                    status: 'complete',
+                  }
+                : file,
+            ),
+          );
+        }),
+      ),
+    ).then(() => {
```

Comment:
I see a `then` function here, but nothing to handle the error state. Is the error state implemented?

Reply:
Added a catch function as well.